src/main/java/world/bentobox/bentobox/managers/IslandWorldManager.java

```diff
@@ -143,6 +155,16 @@ public boolean isKnownFriendlyWorldName(String name) {
                 .anyMatch(gm -> gm.getWorldSettings().getFriendlyName().equalsIgnoreCase(name));
     }
 
+    /**
+     * Associate a world with a game mode. This enables game modes to register more worlds than just the standard
+     * overworld, nether, and end worlds.
+     * @param world world
+     * @param gameMode game mode
+     * @since 1.24.0
+     */
+    public void addWorld(World world, GameModeAddon gameMode) {
+        gameModes.put(world, gameMode);
+    }
```

Comment:
This is required by Boxed, but in theory could be any addon that needs its own worlds for some reason.